File: techon_26/lib/remove_comments.dart
import 'dart:io';

void main() {
  final dir = Directory('c:/Users/Lenovo/OneDrive/Desktop/techon_26/lib');
  
  for (var entity in dir.listSync(recursive: true)) {
    if (entity is File && entity.path.endsWith('.dart') && !entity.path.endsWith('remove_comments.dart')) {
      final lines = entity.readAsLinesSync();
      final newLines = <String>[];
      
      for (var line in lines) {
        if (line.trimLeft().startsWith('//')) {
          continue;
        }
        newLines.add(line);
      }
      
      entity.writeAsStringSync(newLines.join('\n') + '\n');
    }
  }
  print('Comments removed.');
}

File: techon_26/lib/remove_comments.py
import os

dir_path = 'c:/Users/Lenovo/OneDrive/Desktop/techon_26/lib'

for root, _, files in os.walk(dir_path):
    for file in files:
        if file.endswith('.dart'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                lines = f.readlines()
            
            new_lines = []
            for line in lines:
                stripped = line.strip()
                if stripped.startswith('//'):
                    continue
                new_lines.append(line)
            
            with open(filepath, 'w', encoding='utf-8') as f:
                f.writelines(new_lines)

print("Comments removed.")
